### PR TITLE
feat: Added Transit Gateway Attachment tagging

### PR DESCRIPTION
**Pull request by jrkalf-cambrian**
## Description
When creating a VPN gateway using the transit gateway option, it will not put a name to the transit gateway attachment created, nor will it place tags on the transit gateway attachments. This is a shortcoming in the terraform api, or the aws api. This is unknown to me.

In the transit gateway attachment id part of the  [Terraform VPN documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpn_connection#transit_gateway_attachment_id), it's already described that aws_ec2_tag should be used.

I've added the code that should place the proper Name tag to the transit_gateway_attachment object. And if var.tags contains any values, it should run a for_each loop and attach those key/value pairs as well.

## Motivation and Context
Why is it needed, proper tagging of AWS resources and proper visibility of objects in the AWS console.

## Breaking Changes
Not to the best of my knowledge

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
  Updated `example/complete-vpn-connection-transit-gateway` with tagging example to show proof of work.
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
  I've tested and validated against `example/complete-vpn-connection-transit-gateway` with tagging example.

- [x] I have executed `pre-commit run -a` on my pull request
```
 pre-commit run -a
Terraform fmt............................................................Passed
Terraform validate.......................................................Passed
Terraform docs...........................................................Passed
Terraform validate with tflint...........................................Passed
check for merge conflicts................................................Passed
fix end of files.........................................................Passed
```

### CODE REVIEW

**Review by nakopoulos-profile — APPROVED** [review]

**Comment by jrkalf-cambrian** [comment]
I believe we can link this to an issue which was created in the wrong place: https://github.com/terraform-aws-modules/.github/issues/27

It had a similar request.

**Comment by jrkalf-cambrian** [comment]
Is this going to be merged? @antonbabenko

**Review by StephenKing — APPROVED** [review]
Nice one, ran into the same problem. Looks good to me @antonbabenko 
Screenshot 2023-07-18 at 07 01 36

**Comment by StephenKing** [comment]
Certainly, it doesn't apply the [Default Tags of the AWS provider](https://www.hashicorp.com/blog/default-tags-in-the-terraform-aws-provider), but that's more a limitation of the provider (`aws_ec2` resource should do that?), as I guess we can't even access those.

**Comment by jrkalf-cambrian** [comment]
not stale. Please review @antonbabenko

**Comment by jrkalf-cambrian** [comment]
Could you be so kind to review and don't let it auto expire @antonbabenko

File: main.tf [hunk 1]
```diff
@@ -473,3 +473,18 @@ resource "aws_vpn_connection_route" "default" {
 
   destination_cidr_block = element(var.vpn_connection_static_routes_destinations, count.index)
 }
+
+### Fix tagging on Transit Gateway Attachment if it exists
+resource "aws_ec2_tag" "tags" {
+  for_each = { for key, value in merge({ "Name" = local.name_tag }, var.tags) : key => value if var.connect_to_transit_gateway == true }
```
Comment (antonbabenko) [inline main.tf:479]:
```suggestion
  for_each = { for key, value in merge({ "Name" = local.name_tag }, var.tags) : key => value if var.create_vpn_connection && var.connect_to_transit_gateway }
```

**Comment by antonbabenko** [comment]
@jrkalf-cambrian Please fix the failing build and I will merge it.

**Comment by jrkalf-cambrian** [comment]
@antonbabenko ran the `pre-commit run --all-files` to update and commit all updated README.md files.
```
# pre-commit run --all-files
Terraform fmt............................................................Passed
Terraform validate.......................................................Passed
Terraform docs...........................................................Passed
Terraform validate with tflint...........................................Passed
check for merge conflicts................................................Passed
fix end of files.........................................................Passed
```

**Comment by antonbabenko** [comment]
This PR is included in version 3.7.0 :tada:

**Comment by StephenKing** [comment]
> Certainly, it doesn't apply the [Default Tags of the AWS provider](https://www.hashicorp.com/blog/default-tags-in-the-terraform-aws-provider), but that's more a limitation of the provider (`aws_ec2` resource should do that?), as I guess we can't even access those.

actually I recently stumbled over [data.aws_default_tags](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/default_tags)

**Pull request merged**